### DISCUSSION TITLE
Git commit message changed when pushing to GitHub?

### DISCUSSION THREAD

**Discussion by y-chen** [community/community#78337] [Actions] [answered]
### Select Topic Area

Bug

### Body

Hi all,

I have these two simple steps that I use to determinate if the CICD should run or not:

```
- name: Last Commit
  id: last-commit
  run: echo "last_commit=$(git log -1 --format=%B | head -n 1)" >> $GITHUB_OUTPUT

- name: Skip CICD
  id: skip-cicd
  run: echo "value=${{ contains(steps.last-commit.outputs.last_commit, '[skip-cicd]') }}" >> $GITHUB_OUTPUT
```

It just checks if the latests commit message contains `[skip-cicd]` and if it does, the CICD won't run. It works when I merge a PR into the main branch but it doesn't when I push to a feature branch.

As far as I can see it's because when I push  commits to a feature branch, the commit message is changed to something like `Merge 13b3d7ee06fc38766c9443dec492fec6b241fe69 into e0025c356a238acf7913b1feaad34b307d43e6d9`.

This causes the CICD to always run.

Any idea or suggestion regarding this issue?

Thank you.

Best Regards,
Yiyi Chen

**Comment by odinho** [comment, marked as answer]
I have the same issue. GitHub creates an automatic merge commit where they merge `main` back into the branch before running it. So that all changes from `main` are also included in the branch.

This can be useful if the intention is for it to be merged and for the combined result to work well on `main`. Like an optimistic merge queue with less waiting. However, I'm also not thrilled about it, especially that I don't *know* about it, and it seems very hard to get information about what happens.

It should be quite easy to work around though, you can skip this merge commit and go to the first parent. There are many ways to do that, i.e. you could `git checkout HEAD^`, or maybe even `git checkout ${GITHUB_HEAD_REF:-$GITHUB_REF_NAME}` (after a fetch). That will of course build and test other code that that merged with `main` (but this is sort of a wtf if you don't already know it does this).

Or you could handle it in the code by just looking at what you need, i.e. for your example, this should probably fix it:

```diff
 - name: Last Commit
   id: last-commit
-  run: echo "last_commit=$(git log -1 --format=%B | head -n 1)" >> $GITHUB_OUTPUT
+  run: |
+    git fetch
+    echo "last_commit=$(git log -1 --format=%s ${GITHUB_HEAD_REF:-HEAD})" >> $GITHUB_OUTPUT
```

**Reply by y-chen** [reply]
Hey,

sorry, I forgot about this discussion.

I had to add this before

```
      - name: Fetch additional branches
        run: |
          git fetch origin +${{ github.head_ref }}:${{ github.head_ref }}
```

But then it worked like a charm!

Thank you very much! :)

Yiyi
